sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -136,34 +151,42 @@ func (c *Controller) Stop() {
 	}
 }
 
-// We assume that the only metric schedule is one with an inclusion pattern
-// that includes all metrics
-func (c *Controller) OnInitialConfig(config *dynamicconfig.Config) error {
-	err := config.Validate()
+// Called by notifier if we Register with one.
+func (c *Controller) OnInitialConfig(config *notify.Config) error {
+	err := config.ValidateMetricConfig()
 	if err != nil {
 		return err
 	}
 
-	c.period = time.Duration(config.MetricConfig.Schedules[0].Period) * time.Second
+	c.dynamicExtension.SetSchedules(config.MetricConfig.Schedules)
+	c.period = c.updateFromSchedules()
 
 	return nil
 }
 
-func (c *Controller) OnUpdatedConfig(config *dynamicconfig.Config) error {
+// Called by notifier if it receives an update.
+func (c *Controller) OnUpdatedConfig(config *notify.Config) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	err := config.Validate()
+	err := config.ValidateMetricConfig()
 	if err != nil {
 		return err
 	}
 
+	c.dynamicExtension.Clear()
+
+	c.dynamicExtension.SetSchedules(config.MetricConfig.Schedules)
+	c.period = c.updateFromSchedules()
+
 	// Stop the existing ticker
-	// Make a new ticker with the new sampling period
 	c.ticker.Stop()
-	if len(config.MetricConfig.Schedules) > 0 {
-		c.period = time.Duration(config.MetricConfig.Schedules[0].Period) * time.Second
+	// If no schedules, or all schedules have a period of 0, controller never exports.
```

Comment:
William and I decided that having a period of 0 should mean "don't collect". This is because OpenTelemetry has no concept of an arbitrary collection period.